### ISSUE TITLE
how many epoch training need for converge when training streaming conv net?

### ISSUE THREAD

**Issue by mlexplore1122** [facebookresearch/wav2letter#770]
### Question
I am training streaming convnet with 2k hours custom dataset in my language.
But with epoch 60, loss still 7.706, wer train just walk around 50, and loss decrease so slow just 0.2 after approx 20 epoch.
And i have found some log in wave2letter issuse say after epoch 79 with 1k hour let make loss just around 5, and wer around 20. 
I have already test with transfer learning from librivox pretrained in wav2letter. But training slow too.
I have remove saug layer in file .arch and replace with config saug in training, and when i remove saug when training model converge so fast, but test wer and loss not decrease like overfitting. 
So I don't know what is reason for slow training or it's normal when training conv streaming. 

this is my log when training from scratch: 
my [log_v1](https://drive.google.com/file/d/1PrDCaChKnlCrhPkDJ7fhOHLHXGCwQkvH/view?usp=sharing) train from scratch with lr decay schedule and saug -- saug_fmaskf=27, saug_fmaskn=2, saug_tmaskp=0.8, saug_tmaskt=80, saug_tmaskn=2.

my [log_v2 and v3](https://drive.google.com/file/d/1cFNW1kXbX1G9JR0Sc5fqK7cOVmhiV7dm/view?usp=sharing) i fork training from last checkpoint of v1 with lr=0.4 to epoch 14 with same saug like v1 and after that (after epoch 15) i change saug _tmaskt=40 (for faster converge) so loss decrease.

log of my transfer test with 100h. [log_100h](https://drive.google.com/file/d/1tX3TBwIFeczNqtdb-SoHxEsofT7XOufJ/view?usp=sharing). It's use saug_tmaskt=40

#### Additional Context
I train with 6 gpu, and batchsize of each is 16. And 1 epoch need 1.5 hour for complete.
My config is same as config in streaming conv recipe, except difference about lexicon and tokens.
[my_config](https://drive.google.com/file/d/1joCQ_kVc79bUd6WWsf1-wC-DXieIkL74/view?usp=sharing) 
number of tokens in AM: 1220 (like word piece)
So pls tell me what should i do now? or give me some advice.
Thanks you

**Comment by vineelpratap** [comment]
Hi, it looks like you are keeping using Spec Augment twice - from `--flagsfile` and `SAUG` layer in arch file. You should remove one of them (may be SAUG layer in arch file). 

Other things you can try
- reducing the dropout. Replace 0.1 in the [arch file] with 0.05 or 0.0 
(https://github.com/facebookresearch/wav2letter/blob/master/recipes/models/streaming_convnets/librispeech/am_500ms_future_context.arch) with 0.
- Try momentum of 0.8 (--momentum 0.8)


Also, looks like you are using "|" as word separator instead of "_" used in the recipe.  So, make sure your lexicon also reflects this.

**Comment by trangtv57** [comment]
thanks you @vineelpratap,
first: I have removed SAUG layer in origin arch of streaming convnet. I using "|" for word seperator and my lexicon are right for it. And thanks you about suggest momentum, and dropout, i will try and report later.

**Comment by mlexplore1122** [comment]
thanks you @vineelpratap,
first: I have removed SAUG layer in origin arch of streaming convnet. I using "|" for word seperator and my lexicon are right for it. And thanks you about suggest momentum, and dropout, i will try and report later.
additional info: My audio compare with librispeech dataset are: my audio more noise, and in social environment, hard to listen, so is it can be a reason for hard converge?

**Comment by vineelpratap** [comment]
>  my audio more noise, and in social environment, hard to listen, so is it can be a reason for hard converge?

Yeah, convergence depends on the dataset. With SpecAugment convergence could further slowdown as we are sending more "noisy" data to the acoustic model. Also, the valid WER seems to be around 20 which is not terrible.

**Comment by mlexplore1122** [comment]
I compare my valid wer with another setup in transformer s2s end2end, and it's so low then just 10 wer. 
One more question i want ask about size of model when i use fork for transfer learning  like #507 , I don't know when use option fork for transfer my model size approximately 700 Mb then 400 Mb in train from scatch.
